### PR TITLE
[Fix] Error when trying to show preview of undefined filetype

### LINKED ISSUE #6917: Error uploading file when mime type is undefined and size < 10mb.
Rocket.Chat Version: 0.55.1
Running Instances: 1
DB Replicaset OpLog: Disabled
Node Version: v4.8.1

Upload fail when the file type is not recognized by browser and the size is lower than 10 mb.

```
Uncaught TypeError: Cannot read property 'indexOf' of undefined
    at r (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:260:14377)
    at i (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:260:15133)
    at fileUpload (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:260:18133)
    at Object.dropped .dropzone-overlay (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:264:1461)
    at Object.o.(anonymous function) (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:258:620)
    at https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:130:6128
    at Function.e._withTemplateInstanceFunc (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:130:5707)
    at h.View.<anonymous> (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:130:6080)
    at https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:129:27045
    at Object.h._withCurrentView (https://chat.teamto.fr/3ddab15a0c8874398ba3db26458f5d286c68a2ef.js?meteor_js_resource=true:129:23116)
```

The error come from the file rocketchat-ui/client/lib/fileUpload.coffee line 13, `file.file.type` is `undefined`.

I think maybe we should avoid trying to preview the file in this case.

### LINKED ISSUE #6922: Cannot send files
I cannot send some files (eg. `.blender` files) with Rocket.Chat. This bug was found in version `0.55.1`, but is still present in the newest version `0.56.0-rc.1`. There is no error message, I can't find anything interesting in logs, just like nothing happens. Checked on the web client and Ubuntu app (v 2.5.0).
We accept all media types (see screen below), but when I change file extension from `.blend`, to eg. `.jpg`, everything works fine and file can be send.

image

### PR DESCRIPTION
**Pull request by geekgonecrazy**
closes #6922
closes #6917

@RocketChat/core 

Closes #ISSUE_NUMBER

### CODE REVIEW

File: packages/rocketchat-ui/client/lib/fileUpload.coffee [hunk 1]
```diff
@@ -9,6 +9,8 @@ getUploadPreview = (file, callback) ->
 	# If greater then 10MB don't try and show a preview
 	if file.file.size > 10 * 1000000
 		callback(file, null)
+	else if file.file.type is undefined
```
Comment (sampaiodiego) [inline packages/rocketchat-ui/client/lib/fileUpload.coffee]: maybe testing any `falsy` value would be safier.. what do you think? like `else if not file.file.type?`

Comment (geekgonecrazy) [inline packages/rocketchat-ui/client/lib/fileUpload.coffee]: Of course :+1: makes sense.  That's coffeescript for you lol

**Comment by geekgonecrazy** [comment]
@sampaiodiego fixed :+1:

**Comment by staticjinno** [comment]
i'm on 0.57.3. i'm able to upload an extension-less file, but trying to download it will fail in browser & do nothing in app.

**Pull request merged**